core/src/core/transaction.rs

```diff
@@ -36,6 +40,184 @@ use util::static_secp_instance;
 use util::RwLock;
 use util::ToHex;
 
+/// Fee fields as in fix-fees RFC: { future_use: 20, fee_shift: 4, fee: 40 }
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct FeeFields(u64);
+
+impl DefaultHashable for FeeFields {}
+
+impl Writeable for FeeFields {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ser::Error> {
+		writer.write_u64(self.0)
+	}
+}
+
+impl Readable for FeeFields {
+	fn read<R: Reader>(reader: &mut R) -> Result<Self, ser::Error> {
+		let fee_fields = reader.read_u64()?;
+		Ok(Self(fee_fields))
```

Comment:
Should we use `try_from` here? Currently this impl allows us to construct FeeFields with an illegal fee value.

Reply:
the height oblivious reader has no notion of illegal

Reply:
Ok, cleared up after discussion on keybase.